src/textual/_types.py

```diff
@@ -1,12 +1,6 @@
 from typing import TYPE_CHECKING, Any, Awaitable, Callable, List, Union
 
-from typing_extensions import (
-    Literal,
-    Protocol,
-    SupportsIndex,
-    get_args,
-    runtime_checkable,
-)
+from typing_extensions import Protocol
```

Comment:
I have this very vague recollection (probably from a conversation I wasn't fully involved in) that there was a reason why these were imported here, like this, and then imported from here elsewhere in Textual; but I don't recall the detail.

As I look at the code I can't see why this PR *wouldn't* make this change -- it makes sense to be -- but it might be worth double-checking what the original motivation was.

Reply:
Thanks for the heads-up.
I'll merge after I make sure all is good.

Reply:
After talking to Darren, it looks like these are remains of an old pattern we used to follow in our code base where we had plenty of Python version checks to figure out whether to import things from `typing` or `typing_extensions`, and so we wanted to import all those types in a single `_types.py` file so that we could then import from there without version checks.

Given that removing them and “fixing” the imports elsewhere to import directly from `typing_extensions` didn't break anything, we can go ahead with this PR.